init/src/hpx_wrap.cpp

```diff
@@ -23,14 +25,14 @@ namespace hpx_start
     // by this code.
     HPX_SYMBOL_EXPORT extern bool include_libhpx_wrap;
     HPX_SYMBOL_EXPORT bool include_libhpx_wrap __attribute__((weak)) = false;
+    HPX_SYMBOL_EXPORT extern std::string app_name_libhpx_wrap;
```

Comment:
This most likely has to be weak as well. If `hpx_main.hpp` isn't included by the application the name will never be defined, but that's ok since it also won't be used.